source/_posts/2023-09-06-release-20239.markdown

```diff
@@ -134,45 +125,104 @@ Other more advanced fields, like attributes, are not available from the UI. You
 With the preview, you see a live update of the sensors value while you are writing the template.
 
 <p class='img'>
-<img src='https://media.discordapp.net/attachments/427516175237382144/1146448232839397486/SPOILER_CleanShot_2023-08-30_at_16.15.142x.png?width=1398&height=1402'></a>
-Temporary screenshot that needs to be replaced.
+<img src='/images/blog/2023-09/template-sensor.png'>
+Screenshot of creating a template sensor.
 </p>
 
 
-## Actions for template entities
+## Actions for trigger template entities
+
+Trigger template entities can now have an action block that is executed after the trigger but before the entities are rendered.
+
+This will allow users to fetch things like calendar events, weather forecast using services with responses and use them in their template entities. You can even use OpenAI to set the value of your entities now!
+
+All variables from the action block will be available in the template.
+
+An example that checks at midnight if you have something in your calendar tomorrow and sets a binary sensor:
+
+{% raw %}
+
+```yaml
+template:
+  - trigger:
+      - platform: time
+        at: "00:00:00"
+    action:
+      - service: calendar.list_events
+        target:
+          entity_id: calendar.calendar
+        data:
+          start_date_time: "{{ today_at() + timedelta(days=1) }}"
+          duration:
+            hours: 24
+        response_variable: events
+    binary_sensor:
+      - name: Calendar events tomorrow
+        state: "{{ events.events is defined and events.events | count > 0 }}"
+```
+
+{% endraw %}
 
-Trigger template entities can now have an action block that is executed after the trigger but before the entities are rendered. All variables from the action block will be available during the entity rendering. This will allow users to fetch things like weather forecast or calendar events and use them in their template entities. Long term goal is to align the trigger template entities with automation syntax so we can support things like tracing too.
 
 ## Weather forecast service
```

Comment:
I think it would be worth adding some of the context from this post by @bdraco as it may head off the inevitable concerns about this change: https://community.home-assistant.io/t/petition-dont-delete-the-forecast-attribute/609298/150